### DISCUSSION TITLE
switching to using a computer running Linux

### DISCUSSION THREAD

**Discussion by ESH-y** [firemodels/fds#12995] [General]
Hey,
In the company where I work, we are interested in switching to using a computer running Linux. I would appreciate help regarding a few issues:
1. I would like to know what the advantages and disadvantages are in Linux while running FDS.
2. Has anyone who made this transition from Windows to Linux seen an improvement in running performance? (mainly in terms of speed and stability).
3. I would appreciate it if anyone has a computer recommendation? In terms of hardware, amount of processors, amount of memory, etc.
4. Which is better - to run on a Linux computer and communicate with it through a Windows computer or through a Mac computer?
4. I would like to know if there is anyone here who models in pyrosim on a Mac computer? Does it have advantages/disadvantages?

I would really appreciate any help and information on the above matter.

Thanks, have a nice day

**Comment by mcgratta** [comment]
If you are buying a single computer, there is no significant advantage to linux over Windows. The CPU will determine the compute speed. The advantage of linux is when you have a cluster of compute nodes connected by a fast network. Then there are tools like job schedulers (e.g. SLURM) and debuggers that can enable you to run jobs with hundreds of meshes.

**Reply by MNJOLE** [reply]
Hi, may I ask if there is a version of SLURM that you recommend (least error-prone, including compatibility with FDS)?

**Reply by drjfloyd** [reply]
If you are installing slurm, I would recommend just installing whatever the latest version is. If you already have slurm installed and working on your cluster, whatever version you have is likely fine.

**Reply by mcgratta** [reply]
I will add that the only problem we have had at NIST using SLURM with the Intel oneAPI compilers and MPI libraries is that there is some confusion (in our minds anyway) about the use of `srun` or `mpiexe` within the SLURM batch script. `srun` is the preferred way to launch multiple process jobs within the SLURM environment, but over the years we have had some problems using it because `srun` and Intel MPI are not linked by default. There is some environment variable or library to invoke, but the manuals are a bit vague on this. If anyone is reading this and has some wisdom to impart, I am all ears.

**Comment by jfpe37** [comment]
Hi
Following the question above - I would also like to know whether the use of MAC computers is more convenient when using PYROSIM and PATHFINDER?

**Reply by drjfloyd** [reply]
This forum doesn't provide support for the commercial products made by Thunderhead Engineering. Please contact them for support.
